DEV/org.openl.rules/src/org/openl/rules/testmethod/export/FieldDescriptor.java

```diff
@@ -50,14 +61,24 @@ private static List<FieldDescriptor> nonEmptyFieldsForFlatten(IOpenClass type, L
 
             for (Object value : values) {
                 Object fieldValue = value == null ? null : field.get(value, null);
-                if (fieldValue != null && (!field.getType().isArray() || Array.getLength(fieldValue) > 0)) {
-                    if (fieldValue instanceof Collection) {
-                        fieldType = CastToWiderType.defineCollectionWiderType((Collection<?>) fieldValue);
+                String fieldName = field.getDeclaringClass().toString() +
+                        field.getName() +
+                        fieldValue;
+                if (!coveredFields.contains(fieldName)) {
+                    coveredFields.add(fieldName);
+                    if (skipEmptyParameters ? SKIP_EMPTY_PARAMETER_FILTER.test(fieldType, fieldValue)
+                                            : ALL_PARAM_FILTER.test(fieldType, fieldValue)) {

```

Comment:
What is the reason to use `ALL_PARAM_FILTER` which always returns `true` if `skipEmptyParameters == false` ?